### ISSUE TITLE
connect hangs after upgrading to version 2.7.1

### ISSUE THREAD

**Issue by nbrustein** [livekit/client-sdk-js#1335]
### Describe the bug

I have the following component that launches a room. 

When using version 2.7.0 of `livekit-client`, I see the behavior I'm expecting: 

1. Click "Start a Conversation"
2. The ControlBar immediately appears
3. After a few seconds, I hear the agent's voice, the bars start moving, and I can interact speak back to the agent.

In 2.7.1, the behavior is unexpected. See [this screen capture](https://photos.app.goo.gl/Zict4Jt8UvLg18jN7).

1. 0:03 I click "Start a Conversation"
2. In the UI, the outside bars turn right, but they do not start moving. The control bar does not appear. I don't hear anything
3. In the agent console, I can see that the worker is connected to the room and 2 conversation items are created.
4. 0:20 In the agent console, I see "committed agent speech". I still don't hear anything
5. 0:25 In the Browser console, I see "publishing track". The bars start flashing, but not moving up and down. I still don't hear anything
6. 1:20 A message appears on the screen saying "It's quiet..." I dismiss it.
7. 1:32 In the browser console, I see "Resuming signal connection, attempt 0" then "Resuming signal connection". The control bar appears, but my microphone is initially muted. I still don't hear anything.
8. 1:35 In the broswr console, I see "Resumed signal connection"
9. 1:39 I click in the UI to enable my microphone. In the browser console I see "publishing track".
10. 1:46 I start talking. The bars next to the microphone move. 
11. 1:48 In the agent console, I can see conversation items created for user input_audio.
12. 1:48 In the browser, the bars start moving and I can hear the agent talking.

```
function VoiceCoursePracticeRoom({ tokenAttributes }: { tokenAttributes: TokenAttributes }) {
    const [connectionDetails, unsetConnectionDetails, initializeConnectionDetails] = useConnectionDetails({
        tokenAttributes,
    });
    const [agentState, setAgentState] = useState<AgentState>('disconnected');

    return (
        <div data-lk-theme="default" className="grid h-full content-center bg-[var(--lk-bg)]">
            <LiveKitRoom
                token={connectionDetails?.participantToken}
                serverUrl={connectionDetails?.serverUrl}
                connect={!!connectionDetails}
                audio
                video={false}
                onMediaDeviceFailure={onDeviceFailure}
                onDisconnected={() => {
                    unsetConnectionDetails();
                }}
                className="grid grid-rows-[2fr_1fr] items-center"
            >
                <SimpleVoiceAssistant onStateChange={setAgentState} />
                <ControlBar onConnectButtonClicked={initializeConnectionDetails} agentState={agentState} />
                <RoomAudioRenderer />
                <NoAgentNotification state={agentState} />
            </LiveKitRoom>
        </div>
    );
}
```

### Reproduction

For me, all I had to do was use `LiveKitRoom` with 2.7.1, but I imagine there's more to it than that that I don't understand.

### System Info

```shell
System:
    OS: macOS 14.6.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 587.91 MB / 64.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 21.7.3 - ~/.asdf/installs/nodejs/21.7.3/bin/node
    Yarn: 4.5.1 - ~/.asdf/installs/nodejs/21.7.3/bin/yarn
    npm: 10.8.3 - ~/.asdf/plugins/nodejs/shims/npm
  Browsers:
    Chrome: 131.0.6778.86
```

### Severity

blocking an upgrade

**Comment by lukasIO** [comment]
thanks for the report! 

I'm able to reproduce an issue around the changes made in https://github.com/livekit/client-sdk-js/pull/1324 when used in combination with the components framework. Will investigate why this is happening.

**Comment by lukasIO** [comment]
this was a bug that got fixed in https://github.com/livekit/client-sdk-js/pull/1336 and got released just now as part of `v2.7.2`. 

Thanks again for the detailed report!

**Comment by nbrustein** [comment]
Yup. Works for me. Thanks
